eng/Subsets.props

```
@@ -39,10 +39,20 @@
   <PropertyGroup>
     <DefaultSubsets>clr+mono+libs+installer</DefaultSubsets>
     <DefaultSubsets Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'tvOS'">mono+libs+installer</DefaultSubsets>
-    <DefaultSubsetCategories Condition="'$(TargetOS)' == 'WebAssembly'">mono+libs</DefaultSubsetCategories>
+    <DefaultSubsets Condition="'$(TargetOS)' == 'WebAssembly'">mono+libs</DefaultSubsets>
   </PropertyGroup>
 
+  <!-- Init _subSet here in to allow RuntimeFlavor to be set as early as possible -->
   <PropertyGroup>
+    <_subset Condition="'$(Subset)' != ''">+$(Subset.ToLowerInvariant())+</_subset>
+    <_subset Condition="'$(Subset)' == ''">+$(DefaultSubsets)+</_subset>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <RuntimeFlavor Condition="'$(TargetOS)' == 'tvOS' or '$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'WebAssembly'">Mono</RuntimeFlavor>
+    <RuntimeFlavor Condition="'$(RuntimeFlavor)' == '' and ($(_subset.Contains('+mono+')) or $(_subset.Contains('+mono.runtime'))) and (!$(_subset.Contains('+clr+')) and !$(_subset.Contains('+clr.runtime')))">Mono</RuntimeFlavor>
```

Lost some `+` in the move, it looks like, if I understand the pattern right:

```suggestion
    <RuntimeFlavor Condition="'$(RuntimeFlavor)' == '' and ($(_subset.Contains('+mono+')) or $(_subset.Contains('+mono.runtime+'))) and (!$(_subset.Contains('+clr+')) and !$(_subset.Contains('+clr.runtime+')))">Mono</RuntimeFlavor>
```

I would put it on multiple lines for easier reading/review, but this is subjective and I'm sure some people are annoyed at me doing it all the time 😄:

```xml
    <RuntimeFlavor
      Condition="
        '$(RuntimeFlavor)' == '' and
        (
          $(_subset.Contains('+mono+')) or
          $(_subset.Contains('+mono.runtime'))
        ) and
        (
          !$(_subset.Contains('+clr+')) and
          !$(_subset.Contains('+clr.runtime'))
        )">Mono</RuntimeFlavor>
```

---

@steveisok I edited my message on Teams and added the missing pluses :D You probably took mine before the edit :D

---

FWIW, seems like this is a nit as well to me--I think it would only cause a problem if we added a `clr.runtimefoobar` subset.